image/api_selector.go

```diff
@@ -102,16 +103,27 @@ func (as *APISelector) makeImageRequest(urlString string, bodyLines []string) (*
 	b.MaxInterval = 10 * time.Second
 	b.MaxElapsedTime = time.Minute
 
-	err := backoff.Retry(func() (err error) {
+	err := backoff.Retry(func() error {
 		resp, err := http.Post(urlString, imageAPIRequestContentType,
 			strings.NewReader(strings.Join(bodyLines, "\n")+"\n"))
 
 		if err != nil {
 			return err
 		}
 		defer resp.Body.Close()
+
 		responseBody, err = ioutil.ReadAll(resp.Body)
-		return
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode != 200 {
+			return errors.Errorf("expected 200 status code, received status=%d body=%q",
```

Comment:
Would it be a good idea to include from where we expected the 200 status code? Something like "expected 200 status code from Job Board, received…".